doc/CMakeLists.txt

```diff
@@ -308,30 +311,45 @@ add_docset(nrfxlib "${nrfxlib_env}")
 #-------------------------------------------------------------------------------
 # docset: kconfig
 
-# The Kconfig documentation is a separate documentation set that's shared
-# between all modules (nRF, Zephyr, etc.). This makes it possible to link to
-# Kconfig symbols regardless of where they are defined.
-#
-# We rely on the Zephyr Kconfig files pulling in the other Kconfig files, and
-# use the --modules option to gen_kconfig_rest.py to split the documentation
-# into a separate page for each module.
-
-if(WIN32)
-  set(SEP $<SEMICOLON>)
-else()
-  set(SEP :)
-endif()
+file(WRITE ${KCONFIG_BINARY_DIR}/Kconfig.soc.defconfig
```

Comment:
There must be a way to share this with upstream's CMake? @tejlmand?

Reply:
yes, but that requires that the Zephyr module Kconfig code generator is pulled out of here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/kconfig.cmake

and placed in here instead:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/zephyr_module.cmake

The reason it was originally placed in `kconfig.cmake` was because oit was Kconfig code, but maybe it's better to consider this Zephyr module code.

That will make it easier to share with doc generation.
Will make an upstream PR.

Reply:
Actually zephyr_modules.cmake may not be the right location.
Soc roots may be defined by a Zephyr module but doesn't have to be.

Note: will still take a look to find the cleanest solution.

Reply:
Posted: https://github.com/zephyrproject-rtos/zephyr/pull/35859
Feel free to take that into sdk-zephyr as a fromlist, that will allow us to remove the `file(WRITE)` code.